cql/DashboardLibrary.cql

```diff
@@ -249,6 +251,7 @@ define function BiopsyInterpretation(conclusions List<System.Concept>):
         when c in "Histologic CIN3" then {text: 'CIN3', rank: 6}
         when c in "AIS" then {text: 'AIS', rank: 7}
         when c in "Histologic cancer" then {text: 'Cancer', rank: 8}
+        when c ~ "Insufficient" then {text: 'Insufficient/Unsatisfactory', rank: 1}
```

Comment:
You will need to merge or rebase with Mike's branch - https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/153 - since that will require you adding some data to the "Insufficient" Interpretation.

Reply:
Got it. Rebased: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/compare/0327bf4edee10c4e9c9c060365d0d9e99f9b6eaa..e71e80a853012df8e9b1374ab3476d3f42da1684

Reply:
Looks like Mike changed returning object from `Interpretation`. It consists of `riskTableInput`, `label` `rank` and `grade`. Should I change `Insufficient` to fit in there or the current structure `text` and `rank` is fine?

```
        when c ~ "Endometrial Stromal Cells" then {riskTableInput: 'EMSC', label: 'Endometrial stromal cells', rank: 1, grade: highlightGrades.normal}
        when c ~ "Unsatisfactory" then {riskTableInput: 'UNSAT', label: 'Unsatisfactory', rank: 1, grade: highlightGrades.normal}
        when c ~ "Absent Transformation Zone" then {riskTableInput: 'ATZ', label: 'Absent Transformation Zone', rank: 1, grade: highlightGrades.normal}
        when c ~ "Insufficient" then {text: 'Insufficient/Unsatisfactory', rank: 1}

```

Reply:
Yeah, you will want to update the "Insufficient" return to match the new style that Mike added. I would imagine this is the main place where you saw merge conflicts. I would recommend adding the following new values for the Insufficient entry:
```suggestion
        when c ~ "Insufficient" then {riskTableInput: 'Insufficient', label: 'Insufficient/Unsatisfactory', rank: 1, grade: highlightGrades.unrecognized}
```
Reasoning:
- `riskTableInput`: Insufficient is not in the risk tables, so the actual text here doesn't matter too much. But this could be used to identify this result if we get around to adding specific recommendations for it.
- `label`: This is what will display in the Dashboard. Our pilot partner indicated they wanted to see "Insufficient/Unsatisfactory" for this.
- `rank`: This determines how high of a priority this result takes over other biopsy results. Essentially, when determining what the final result is for a test that may have multiple results, we take the one with the highest rank, which generally corresponds to the 'most severe' result. I agree with your choice to give Insufficient a rank of 1, since if any other known result was present, we would want that result to take precedence over the "Insufficient" one.
- `grade`: This determines the text color and shading that you will see in the Dashboard. I would suggest adding the unrecognized shading here, since our CDS still will not be able to actually process these "Insufficient" results.

Reply:
Yes, also changed text to "Insufficient/Unrecognized" as discussed at the meeting.

https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/154/commits/f1255600e0d3aba4b6f4bf616af3822cf17236a7

Reply:
Updated format for the "Insufficient" case is consistent with the others now! But I didn't pick up on the request to change "Unsatisfactory" to "Unrecognized"